### ISSUE TITLE
Cache preprocessed data

### ISSUE THREAD

**Issue by cbarkr** [shabbiryusufali/vole#35]
# Background
- 99% of `train.py`'s runtime is preprocessing 
- Binary -> CFG -> normalization -> IR embedding -> converting to `torch_geometric.data.Data` is a lot of work!
- Assuming the `list[torch_geometric.data.Data]`, can be serialized nicely (unlike the CFGs themselves...), [`torch.save`](https://docs.pytorch.org/docs/stable/generated/torch.save.html) can be used to achieve this end
- We could potentially determine whether or not to discard the cache and recompute the results by using [`pathlib.Path.stat`](https://docs.python.org/3/library/pathlib.html#pathlib.Path.stat) and comparing the [`st_mtim`](https://www.man7.org/linux/man-pages/man3/stat.3type.html) to the current time

# Deliverables
- Cache results to `.vole_cache` using the above spec
